### PR TITLE
Create loader for login and get counties

### LINKED ISSUE #44: Loaders
Show loaders for some network requests:
- login
- get counties

TBD - manual sync

### PR DESCRIPTION
**Pull request by henriquehorbovyi**
In this PR I've created a `Result` sealed class to handle data from requests. Also I've created a `Dialog Fragment` to be shown when something is being loaded in background.

closes #44

### CODE REVIEW

File: app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginActivity.kt [hunk 1]
```diff
@@ -29,13 +40,49 @@ class LoginActivity : BaseActivity<LoginViewModel>() {
...
+                },
+                onFailure = {
+                    progressDialog.dismiss()
+                    Snackbar.make(loginButton, "Something went wrong!", Snackbar.LENGTH_SHORT)
+                        .show()
+                },
+                onLoading = {
+                    progressDialog.show(supportFragmentManager, ProgressDialogFragment.tag)
+                }
+            )
```
Comment (AlexandraDaraban) [inline app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginActivity.kt]: The only difference between these is the next activity that should be started if the user logged in successfully. You could implement a method that has the activity as a parameter.

File: app/src/main/java/ro/code4/monitorizarevot/widget/ProgressDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_progress_dialog, container, false)
+    }
+
+    companion object {
+        const val tag = "ProgressDialogFragment"
```
Comment (AlexandraDaraban) [inline app/src/main/java/ro/code4/monitorizarevot/widget/ProgressDialogFragment.kt]: You could use `ProgressDialogFragment::class.java.simpleName` and the variable name should be all uppercase.

File: app/src/main/res/layout/fragment_progress_dialog.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/loading_label"
+        android:textSize="16sp"
+        android:layout_marginEnd="24dp"
+        app:layout_constraintBottom_toBottomOf="@+id/progressBar"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/progressBar"
+        app:layout_constraintTop_toTopOf="@+id/progressBar" />
```
Comment (AlexandraDaraban) [inline app/src/main/res/layout/fragment_progress_dialog.xml]: Please apply one of the styles for the `TextView` and use dimensions from the `dimens.xml` for setting margins/paddings

File: app/src/main/res/layout/fragment_progress_dialog.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    <ProgressBar
+        android:id="@+id/progressBar"
+        style="?android:attr/progressBarStyle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_margin="24dp"
```
Comment (AlexandraDaraban) [inline app/src/main/res/layout/fragment_progress_dialog.xml]: Please use dimensions from the `dimens.xml` for setting margins/paddings.

File: app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginActivity.kt [hunk 5]
```diff
@@ -11,16 +14,24 @@ import ro.code4.monitorizarevot.helper.startActivityWithoutTrace
 import ro.code4.monitorizarevot.ui.base.BaseActivity
 import ro.code4.monitorizarevot.ui.branch.BranchActivity
 import ro.code4.monitorizarevot.ui.onboarding.OnboardingActivity
+import ro.code4.monitorizarevot.widget.ProgressDialogFragment
 
 class LoginActivity : BaseActivity<LoginViewModel>() {
 
+    private val progressDialog: ProgressDialogFragment by lazy { ProgressDialogFragment() }
```
Comment (AlexandraDaraban) [inline app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginActivity.kt:17]: The `dialog` should be dismissed if the activity is dismissed/closed (e.g: the user dismisses the app while a dialog is shown and a `java.lang.IllegalArgumentException: View not attached to window manager` exception is thrown)

**Review by AlexandraDaraban — CHANGES_REQUESTED** [review]
Hi @henrikhorbovyi, the PR looks good but there are some issues that I would like you to address. Also, there are some conflicts that also need to be resolved. Thanks!

**Comment by henriquehorbovyi** [comment]
I'm finishing your change requests, but the worse thing is that, there are conflicts and I should understand how to fix it to don't make a messy :smile:

**Comment by henriquehorbovyi** [comment]
@AlexandraDaraban I've fixed what you proposed. In this commit  ac61810570f334d58446563f48d650cb35af816d you can check the last changes, cause I had to fix some conflicts.

**Review by AlexandraDaraban — APPROVED** [review]

**Pull request merged**